tensorflow/cc/gradients/nn_grad.cc

```diff
@@ -47,6 +48,38 @@ Status SoftmaxGrad(const Scope& scope, const Operation& op,
 }
 REGISTER_GRADIENT_OP("Softmax", SoftmaxGrad);
 
+Status SoftmaxCrossEntropyWithLogitsGrad(const Scope& scope,
+                                          const Operation& op,
+                                          const std::vector<Output>&
+                                          grad_inputs,
+                                          std::vector<Output>* grad_outputs) {
+  // Softmax gradient with cross entropy logits function
+  // We multiply the backprop for cost with the gradients - op.output[1]
+  // There is no gradient for labels
+
+  auto softmaxGrad = op.output(1);
+  auto gradLoss = grad_inputs[0];
+  auto gradGrad = grad_inputs[1];
+
+  auto tempGrad = Mul(scope, gradLoss, softmaxGrad);
+
+  // TODO: Check if this sufficient, need a check for null ?
+  if (gradGrad.op().output_type(0) != 0) {
+	  auto logits = op.input(0);
+	  auto softmax = ops::Softmax(scope, logits);
+	  auto prod = ops::MatMul(scope, gradGrad, softmax);
+	  auto squeezeProd = ops::Squeeze(scope, prod);
+	  auto fProd = Sub(scope, gradGrad, squeezeProd);
```

Comment:
it seems this is missing the final `* softmax` before `Add` and after `Sub` there needs to be a `*` of `softmax`. What does `*` mean in this context? Element wise mul? Perhaps this is the difference.

Reply:
cool ! This might be the issue. I lost track of the Python changes. Sorry about that.  I will check this and update soon. Thanks again !